src/main/scala/ai/privado/tagger/S3Tagger.scala

```diff
@@ -0,0 +1,130 @@
+package ai.privado.tagger
+
+import ai.privado.cache.{DatabaseDetailsCache, S3DatabaseDetailsCache}
+import ai.privado.languageEngine.java.language.NodeStarters
+import ai.privado.model.DatabaseDetails
+import io.shiftleft.codepropertygraph.generated.{Cpg, Languages, Operators}
+import org.slf4j.LoggerFactory
+import io.shiftleft.codepropertygraph.generated.nodes.{Call, Method}
+import io.shiftleft.semanticcpg.language.*
+import io.shiftleft.codepropertygraph.generated.nodes.JavaProperty
+import org.slf4j.LoggerFactory
+import overflowdb.traversal.*
+
+import scala.collection.mutable.ListBuffer
+
+class S3Tagger(cpg: Cpg) extends PrivadoSimpleCpgPass(cpg) {
```

Comment:
Any reason to use PrivadoSimpleCpgPass instead of Concurrent pass?

Reply:
There is no clear parallelization we can do there I think 🤔 There are only three types of rules with different logic that runs linearly for them. We can think about optimizing this later IMO